dpp/src/state_transition/state_transition_types.rs

```diff
@@ -1,10 +1,19 @@
-use anyhow::bail;
-use std::convert::TryFrom;
+use num_enum::{IntoPrimitive, TryFromPrimitive};
 
 use serde_repr::{Deserialize_repr, Serialize_repr};
 
 #[repr(u8)]
-#[derive(Serialize_repr, Deserialize_repr, PartialEq, Eq, Clone, Copy, Debug)]
+#[derive(
+    Serialize_repr,
+    Deserialize_repr,
+    PartialEq,
+    Eq,
+    Clone,
+    Copy,
+    Debug,
+    TryFromPrimitive,
```

Comment:
Oh, didn't know about such a trait, cool! What error it gives if it can't be coverted?

Reply:
The macro comes from an external crate `num_enum`. I was tired of implementing it manually ;)
Example of generated error ` No discriminant in enum `StateTransitionType` matches the value `10`` Ideally it should also show possible values